src/client/async_client.rs

```
@@ -1,65 +1,572 @@
+
 /// A [`Client`] facilitates interactions with the PostHog API over HTTP.
 pub struct Client {
     options: ClientOptions,
     client: HttpClient,
+    local_evaluator: Option<LocalEvaluator>,
+    _flag_poller: Option<AsyncFlagPoller>,
+    /// Tracks which feature flags have been called for deduplication.
+    /// Maps distinct_id -> set of feature flag keys that have been reported.
+    distinct_ids_feature_flags_reported: Arc<RwLock<HashMap<String, HashSet<String>>>>,
 }
 
 /// This function constructs a new client using the options provided.
 pub async fn client<C: Into<ClientOptions>>(options: C) -> Client {
-    let options = options.into();
+    let mut options = options.into();
+    // Ensure endpoint_manager is properly initialized based on the host
+    options.endpoint_manager = EndpointManager::new(options.host.clone());
     let client = HttpClient::builder()
         .timeout(Duration::from_secs(options.request_timeout_seconds))
         .build()
         .unwrap(); // Unwrap here is as safe as `HttpClient::new`
-    Client { options, client }
+
+    let (local_evaluator, flag_poller) = if options.enable_local_evaluation {
+        if let Some(ref personal_key) = options.personal_api_key {
+            let cache = FlagCache::new();
+
+            let config = LocalEvaluationConfig {
+                personal_api_key: personal_key.clone(),
+                project_api_key: options.api_key.clone(),
+                api_host: options.endpoints().api_host(),
+                poll_interval: Duration::from_secs(options.poll_interval_seconds),
+                request_timeout: Duration::from_secs(options.request_timeout_seconds),
+            };
+
+            let poller = AsyncFlagPoller::new(config, cache.clone());
+            poller.start().await;
+
+            (Some(LocalEvaluator::new(cache)), Some(poller))
+        } else {
+            eprintln!("[FEATURE FLAGS] Local evaluation enabled but personal_api_key not set");
```

We should never hit this case - it should be impossible to set enable_local_evaluation to `true` without having also provided a personal api key, and the client options builder should return an error from `build` if that does happen. 


Aside from that, logging straight to eprintlin is also a totally inappropriate thing to do - we should either hook into tracing, or fail silently, but in practice as mentioned above, this invalid state should be unreachable, and you should be able to safely `unwrap` on the key in the present of `enable_local_evaluation`